configure.ac

```diff
@@ -248,24 +272,29 @@ else
 fi
 
 # Checks for programs.
+if test "x$INSTALL_JUDGEHOST" != xno; then
 AC_PROG_CXX
 AC_PROG_CC
 AC_PROG_CXXCPP
 AC_PROG_CPP
+fi
+
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
 AC_PROG_MKDIR_P
 
 # Check option to build documentation. This option is provided to
 # allow disabling it.
-AC_ARG_ENABLE(doc-build,AS_HELP_STRING([--enable-doc-build],
-[enable building documentation (default: yes).]))
-
-if test "x$enable_doc_build" = "xno"; then
-	AC_SUBST(DOC_BUILD_ENABLED,[no])
-else
-	AC_SUBST(DOC_BUILD_ENABLED,[yes])
+if test "x$INSTALL_DOMSERVER" != xno; then
```

Comment:
I don't think docs should be a subset of the domserver configure target, but a separate target.

Also would be good to keep the flags and variables for docs and domserver/judgehost consistent.